pkg/util/pods.go

```diff
@@ -323,6 +316,48 @@ func podJobIsFailed(pod *corev1.Pod) bool {
 	return false
 }
 
+// checkPendingPeriodic continually calls checkPending
+// After each verification is performed based on the value loaded from the
+// pointer, the timer is reset based on the result or an error is returned.
+// This function only returns when `done` is signaled or an error occurs; for
+// the latter case, the pod which caused the failure is also returned.
+func checkPendingPeriodic(
+	done <-chan struct{},
+	timeout time.Duration,
+	pod *atomic.Pointer[corev1.Pod],
+) (*corev1.Pod, error) {
+	timer := time.NewTimer(0)
+	for {
+		select {
+		case <-done:
+			if !timer.Stop() {
+				<-timer.C
+			}
+			return nil, nil
+		case <-timer.C:
+			// This is based on the invariant that the time point at which the
+			// verification should be performed only moves forward with changes
+			// in pod status.  Whenever the timer expires, the latest version
+			// of the pod is examined and one of two cases can occur:
+			// - No containers have have started/finished during the waiting
+			//   period.  Since the period was the pending timeout, this means
```

Comment:
```suggestion
			//   period. Since the period was the pending timeout, this means
```

Reply:
In general, I see two spaces are used in this comment if there is a new sentence. Is there a convention I know nothing about?

Reply:
You can read https://en.wikipedia.org/wiki/Sentence_spacing if you really want to know.

The short answer is that's how I have `vim` configured the typographical convention I tend to use (unconsciously, mostly), but it could be changed if necessary.

Reply:
I do not care that much. I was just curious.